core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -795,7 +796,8 @@ private[spark] object JsonProtocol {
       case `taskKilled` => TaskKilled
       case `executorLostFailure` =>
         val executorId = Utils.jsonOption(json \ "Executor ID").map(_.extract[String])
-        ExecutorLostFailure(executorId.getOrElse("Unknown"))
+        val reason = Utils.jsonOption(json \ "Loss Reason").map(_.extract[String])
+        ExecutorLostFailure(executorId.getOrElse("Unknown"), reason.getOrElse("Unknown"))
```

Comment:
(I'll add a test).

About `taskEndReasonFromJson`, this line we are commenting on a part of `taskEndReasonFromJson` but let me know if I am missing something:-)

And, I am using Utils.jsonOption() could you please clarify?